app/src/main/java/ar/scacchipa/twittercloneapp/repository/CredentialRepository.kt

```diff
@@ -0,0 +1,43 @@
+package ar.scacchipa.twittercloneapp.repository
+
+import android.content.SharedPreferences
+import ar.scacchipa.twittercloneapp.data.Credential
+import ar.scacchipa.twittercloneapp.utils.Constants
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+class CredentialRepository(
+    private val credentialLocalSource: SharedPreferences,
```

Comment:
Also maybe the val should be of a interface type, an interface hiding the datasource methods, so you can make unit tests of this repository.

Reply:
I would recommend you not to depend on SharedPreferences directly, but to have some interface LocalStorage, so you can have your SharedPrefsLocalStorage being used in the code and TestLocalStorage in the tests. SharedPrefsLocalStorage will use SharedPreferences under the hood, and TestLocalStorage some Map implementation. Te dejo un link con codigo de ejemplo. https://stackoverflow.com/questions/48728748/how-to-test-sharedpreference-that-inject-in-repository-in-android-mvp-clean-arch

Reply:
It's a excellent article. I implemented it.